src/Network/HTTP2/Client.hs

```diff
@@ -283,7 +283,7 @@ type PushPromiseHandler =
 --
 -- Please refer to 'StreamStarter' and 'StreamDefinition' for more.
 withHttp2Stream :: Http2Client -> StreamStarter a
-withHttp2Stream = _startStream
+withHttp2Stream c s = _startStream c s
```

Comment:
Why was this done? I'd imagine this hurts GHC's ability to do inlining and if you are explicitly wanting to stop inlining, the pragma would be a better way to accomplish that.

Reply:
See this bit of the release notes:

> GHC now implements simplified subsumption, as described in GHC Proposal #287. This change simplifies the type system, and prevents the possibility of GHC silently changing the semantics of user programs, but it does mean that some libraries may need eta-expansion to typecheck. More info here: Subsumption.